### DISCUSSION TITLE
R - Analysis for data separated by comma

### DISCUSSION THREAD

**Discussion by absoluteon** [teamalgoritma/community#79] [Q&A] [answered]
Hi,

I'm trying to analyze the cast for this dataset.  As the cast column's data are separated by comma, is there any way I can do a sort of separate and group back together so I can further analyze?
![image](https://user-images.githubusercontent.com/101559412/166886027-397067cf-0f48-4d44-8b40-49a527cafd4b.png)

I read that for Python we can use multilabel binarizer (found here - point 3.3 https://michael-fuchs-python.netlify.app/2019/06/16/types-of-encoder/#multilabelbinarizer) however I have not found a way to do this in R.

Your advice would be much appreciated, thank you.

**Comment by onlyphantom** [comment, marked as answer]
It looks like you can use `separate(df, cast, c("cast_a", "cast_b"))` where `cast_a` and `cast_b` are the new columns. You can have as many columns as you like. 

If you like the `dplyr`'s `%>%` (margrittr) workflow, it looks like this:

```
library(dplyr)
df <- data.frame(x = c(NA, "x.y", "x.z", "y.z"))
df %>% separate(x, c("A", "B"))
#>      A    B
#> 1 <NA> <NA>
#> 2    x    y
#> 3    x    z
#> 4    y    z
```

Also some good resources: 
1. https://tidyr.tidyverse.org/reference/separate.html
2. `stringr`'s `str_split`: https://www.rdocumentation.org/packages/stringr/versions/1.4.0/topics/str_split

**Reply by absoluteon** [reply]
Think I can work with this, thanks!
